cql3/statements/create_keyspace_statement.cc

```diff
@@ -183,10 +183,9 @@ std::vector<sstring> check_against_restricted_replication_strategies(
             rs_fail_list.emplace_back(locator::replication_strategy_type::simple);
         } else if (simple_strategy_restriction == db::tri_mode_restriction_t::mode::WARN) {
             rs_warn_list.emplace_back(locator::replication_strategy_type::simple);
-        } else if (auto &topology = qp.proxy().get_token_metadata_ptr()->get_topology();
-                topology.get_datacenter_endpoints().size() > 1) {
+        } else if (qp.proxy().get_token_metadata_ptr()->get_datacenter_token_owners().size() > 1) {
             // Scylla was configured to allow SimpleStrategy, but let's warn
-            // if it's used on a cluster which *already* has multiple DCs:
+            // if it's used on a cluster which *already* has multiple DCs with token owners:
```

Comment:
Arguably, they should be using NTS when they have multi-DC setup, whether or not each DC has a token owner.
So previous version of this logic was also fine.

Reply:
Moreover, these two versions are equivalent. If a DC contains only zero-token nodes, `token_metadata::get_datacenter_token_owners` returns a map that contains the name of this DC as a key. I changed `token_metadata::get_datacenter_token_owners` at some point when I worked on this PR and I forgot to revert this change. I'll do it, I like the previous version more.